### DISCUSSION TITLE
Trying to upload a file in Java using webclient, but keep getting errors

### DISCUSSION THREAD

**Discussion by sina2han** [community/community#109151] [Programming Help]
Hello, I am new to github and not sure if this is the right area to ask a question, but i need help in solving an issue in code while uploading a file using webclient in Java as i keep getting errors. The code I tried:
 ```
var file = new File(tempDirectory + File.separator + myFile);

    var multipartBodyBuilder = new MultipartBodyBuilder();
    multipartBodyBuilder.part("ticket", ticket, MediaType.TEXT_PLAIN);
    multipartBodyBuilder.part("file", new FileSystemResource(filePath), MediaType.APPLICATION_PDF).filename(fileName);
    
    var mimeMultipartData = multipartBodyBuilder.build();
HttpHeaders headers = new HttpHeaders();
		headers.setContentType(MediaType.MULTIPART_FORM_DATA);
WebClient webclient = client.getConnectionProvider().createWebClientBuilder(url, true, headers)
				.build();
    return webclient.post().body(BodyInserters.fromMultipartData(builder.build())).exchangeToMono(response -> {
			if (response.statusCode().isError()) {
				return response.createException().flatMap(Mono::error);
			}
			return response.bodyToMono(String.class);
		}).block().trim();

}
```

This returns error 'end of file never reached'. I have tried several solutions like converting to byte etc but nothing worked. Is there something missing or need to be removed from this code extract to make it working. Thanks.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community and welcome to GitHub, @sina2han ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
